src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellDesign.java

```diff
@@ -188,49 +229,85 @@ private Cell registerCell(Cell cell) {
 
 		cell.setDesign(this);
 		cellMap.put(cell.getName(), cell);
+		
+		// add all internal nets when a macro is added to the design
+		if (cell.isMacro()) {
+			for (Cell internal : cell.getInternalCells()) {
+				internalCellMap.put(internal.getName(), internal);
+				internal.setDesign(this);
+			}
+			cell.getInternalNets().forEach(this::addNet);
+		}
+		
 		return cell;
 	}
 
 	/**
-	 * Disconnects and removes the specified cell from this design.
-	 *
-	 * The cell should not be a part of a molecule.
+	 * Disconnects and removes the specified cell from this design. For macro cells,
+	 * all internal nets and cells are also removed from the design. Internal cells
+	 * cannot be removed from the design, an exception will be thrown. Instead,
+	 * remove the parent macro cell. 
 	 *
 	 * @param cell the cell in this design to remove
 	 */
 	public void removeCell(Cell cell) {
 		Objects.requireNonNull(cell);
 		if (cell.getDesign() != this)
 			throw new Exceptions.DesignAssemblyException("Cannot remove cell not in the design.");
-
+		if (cell.isInternal()) 
+			throw new Exceptions.DesignAssemblyException("Cannot remove internal cell from the design. Remove the macro parent cell.");
+		
 		removeCell_impl(cell);
 	}
 
 	private void removeCell_impl(Cell cell) {
-		disconnectCell_impl(cell);
+		
 		cellMap.remove(cell.getName());
 		cell.clearDesign();
+		
+		// remove all of the internal cells and nets if a macro cell is removed
+		if (cell.isMacro()) {
+			for (Cell iCell: cell.getInternalCells()) {
+				iCell.clearDesign();
+				unplaceCell_impl(iCell);
+				internalCellMap.remove(iCell.getName());
+			}
+			cell.getPins().stream().filter(CellPin::isConnectedToNet).forEach(p -> p.getNet().disconnectFromPin(p));
+			cell.getInternalNets().forEach(this::removeNet_impl);
+		}
+		else {
+			disconnectCell_impl(cell);
+		}
 	}
 
 	/**
 	 * Disconnects without removing the specified cell from this design.  This is
-	 * accomplished by unplacing the cell and disconnecting all of its pins.
-	 *
-	 * The cell should not be a part of a molecule.
+	 * accomplished by unplacing the cell and disconnecting all of its pins. For macro
+	 * cells, all internal cells are unplaced. Internal cells cannot be disconnected, an
```

Comment:
Reads weirdly.  Try "Internal cells cannot be disconnected and if attempted, this method will throw a DesignAssemblyException.